File: notebookspython/20-string-rotation.py
def is_fliped_string(s1: str, s2: str) -> bool:
    """Return True if s2 is a rotation of s1."""
    if not isinstance(s1, str) or not isinstance(s2, str):
        raise TypeError("Inputs must be strings.")

    if len(s1) != len(s2):
        return False

    if not s1:
        return True

    return s2 in (s1 + s1)


if __name__ == "__main__":
    assert is_fliped_string("waterbottle", "erbottlewat") is True
    assert is_fliped_string("abcde", "cdeab") is True
    assert is_fliped_string("abcde", "abced") is False
    assert is_fliped_string("", "") is True
    print("All built-in tests passed.")

File: notebookstypescript/20-string-rotation.ts
function isFlipedString(s1: string, s2: string): boolean {
    if (typeof s1 !== "string" || typeof s2 !== "string") {
        throw new TypeError("Inputs must be strings.");
    }

    if (s1.length !== s2.length) {
        return false;
    }

    if (s1.length === 0) {
        return true;
    }

    return s2.includes(s1 + s1);
}

console.log(isFlipedString("waterbottle", "erbottlewat"));
console.log(isFlipedString("abcde", "cdeab"));
console.log(isFlipedString("abcde", "abced"));
